### PR TITLE
use camelCase in EdgeDevice CR

### PR DESCRIPTION
**Pull request by ydayagi**
https://issues.redhat.com/browse/ECOPROJECT-253
Signed-off-by: Yaron Dayagi <ydayagi@redhat.com>

### CODE REVIEW

**Review by jakub-dzon — CHANGES_REQUESTED** [review]
Please move the whole hardware mapping code to a separate, testable package and add tests for it.

**Comment by ydayagi** [comment]
> Please move the whole hardware mapping code to a separate, testable package and add tests for it.

what is the reason for that? what should we test?

**Comment by ydayagi** [comment]
> Please move the whole hardware mapping code to a separate, testable package and add tests for it.

done

File: internal/hardware/hardware_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var (
+	logger       logr.Logger = log.FromContext(context.Background())
+	testHostname string      = "the-best-host"
+)
+
+func TestNilInput(t *testing.T) {
```
Comment (jakub-dzon) [inline internal/hardware/hardware_test.go:18]: Please use Ginkgo for writing tests and use `// given - // when - // then convention` to clearly show what is the setup, what is the actually tested call and what is the verification . See https://github.com/jakub-dzon/k4e-operator/tree/main/internal/utils.

File: internal/hardware/hardware.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/jakub-dzon/k4e-operator/api/v1alpha1"
+	"github.com/jakub-dzon/k4e-operator/models"
+)
+
+func MapHardware(logger logr.Logger, hardware *models.HardwareInfo) *v1alpha1.Hardware {
```
Comment (jakub-dzon) [inline internal/hardware/hardware.go:11]: `logger` is not needed.

File: internal/hardware/hardware_test.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	assert.Nil(t, result)
+}
+
+func TestNilFields(t *testing.T) {
+	input := models.HardwareInfo{}
+	result := MapHardware(logger, &input)
+	assert.NotNil(t, result)
+}
+
+func TestAllMappings(t *testing.T) {
```
Comment (jakub-dzon) [inline internal/hardware/hardware_test.go:29]: Please add tests for non-zero values.

**Review by jakub-dzon — CHANGES_REQUESTED** [review]
The CI build failed due to `internal/hardware` package tests failing; please fix it.
